File: data/derived_cot/rq1_traces/translation/r1/py_to_cpp/code_output/BalancedBrackets.cpp
#include <vector>
#include <string>
#include <stdexcept>

class BalancedBrackets {
public:
    std::vector<char> stack;
    std::string left_brackets;
    std::string right_brackets;
    std::string expr;

    BalancedBrackets(std::string expr) : expr(expr) {
        left_brackets = "({[";
        right_brackets = ")}]";
    }

    void clear_expr() {
        std::string cleaned;
        for (char c : expr) {
            if (left_brackets.find(c) != std::string::npos || 
                right_brackets.find(c) != std::string::npos) {
                cleaned += c;
            }
        }
        expr = cleaned;
    }

    bool check_balanced_brackets() {
        clear_expr();

        for (char Brkt : expr) {
            if (left_brackets.find(Brkt) != std::string::npos) {
                stack.push_back(Brkt);
            } else {
                if (stack.empty()) {
                    throw std::out_of_range("pop from empty stack");
                }
                char Current_Brkt = stack.back();
                stack.pop_back();

                if (Current_Brkt == '(') {
                    if (Brkt != ')') {
                        return false;
                    }
                }
                if (Current_Brkt == '{') {
                    if (Brkt != '}') {
                        return false;
                    }
                }
                if (Current_Brkt == '[') {
                    if (Brkt != ']') {
                        return false;
                    }
                }
            }
        }

        if (!stack.empty()) {
            return false;
        }
        return true;
    }
};

File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_py/code_output/BalancedBrackets.py
class BalancedBrackets:
    def __init__(self, expr: str):
        self.stack = []
        self.leftBrackets = "({["
        self.rightBrackets = ")}]"
        self.expr = expr

    def clearExpr(self):
        cleaned = []
        for c in self.expr:
            if c in self.leftBrackets or c in self.rightBrackets:
                cleaned.append(c)
        self.expr = "".join(cleaned)

    def checkBalancedBrackets(self):
        self.clearExpr()
        for brkt in self.expr:
            if brkt in self.leftBrackets:
                self.stack.append(brkt)
            else:
                if not self.stack:
                    return False
                current = self.stack.pop()
                if current == '(' and brkt != ')':
                    return False
                if current == '{' and brkt != '}':
                    return False
                if current == '[' and brkt != ']':
                    return False
        return not self.stack

    def getExpr(self):
        return self.expr


if __name__ == "__main__":
    b = BalancedBrackets("a(b)c")
    print(b.checkBalancedBrackets())